FilecoinFoundationWeb/filecoin-foundation
# Refactor API Folder and Add Alias [skip percy]

## 📝 Description

This pull request introduces two key changes to improve the organization and clarity of the project's folder structure. 

1. The `api` folder has been renamed to `_api` to align with the existing naming convention for internal namespaces, such as `_components`, `_utils`, etc.
2. A new path alias `@/api/*` has been added to the `tsconfig.json` file, pointing to `src/app/_api/*`. This simplifies imports and maintains consistency across the codebase.

### Motivation
The change ensures that:
- The folder structure remains consistent.
- Developers can clearly identify internal namespaces.
- Imports from the `_api` folder are easier to manage and understand.

- **Type:** Refactor

## 🛠️ Key Changes

- Renamed the `api` folder to `_api` in the `src/app` directory.
- Updated `tsconfig.json` to include a new alias `@/api/*` pointing to `src/app/_api/*`.
- Refactored existing imports to use the new alias `@/api/*` for consistency.

### **Setup**
1. Checkout the branch `refactor/folder-structure-api-alias`.
2. Ensure all environment variables are correctly set (e.g., API keys in `.env`).

### **Steps to Test**
1. Run the project locally (`npm run dev`) and ensure all API endpoints still function as expected.
2. Verify that imports from `@/api/*` resolve correctly in the following locations:
   - Services using `_api`.
   - Hooks dependent on API logic.
   - Any components/pages directly referencing `_api` functions.
3. Build the project (`next build`) and ensure no errors occur.

### **Expected Results**
- API functionality remains unaffected.
- No unresolved imports or build errors.
- All tests pass successfully.

## 🔖 Resources

- [Next.js Path Aliases Documentation](https://nextjs.org/docs/advanced-features/module-path-aliases)
- [TypeScript Paths Documentation](https://www.typescriptlang.org/tsconfig#paths)
